src/CRUD/scan.ts

```diff
@@ -9,28 +16,27 @@ export default async function getAll({
   indexName,
   options = {},
   pagination = false,
-  verbose = false
-}) {
-  let params;
+  verbose = false,
+}: IDLScan) {
+  const params = {
+    TableName: tableName,
+    ...(indexName && { IndexName: indexName }),
+    ...options,
+  };
   try {
-    params = {
-      TableName: tableName,
-      ...(indexName && { IndexName: indexName }),
-      ...options
-    };
     if (verbose) {
       console.log("params", params);
     }
-    let result;
-    let shouldKeepFetching;
+    let result = { Count: 0 };
```

Comment:
The change here broke the `scan` function. The scan function relies on the `result` to be undefined in the first loop of concatenation, otherwise, it'd throw null pointer because `result` has no `Items`